### PR TITLE
Split real and fake clocks

### PR DESCRIPTION
**Pull request by dashpole**
All testing was for the fake clock, so that file is now in the testing sub-dir

/assign @apelisse 
cc @thockin

### CODE REVIEW

File: clock/testing/fake_clock_test.go [hunk 1]
```diff
@@ -14,7 +14,7 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-package clock
+package testing
```
Comment (apelisse) [inline clock/testing/fake_clock_test.go]: we don't have any test for the real clock? I guess it's not that easy to test ...

File: clock/testing/fake_clock.go [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+limitations under the License.
+*/
+
+package testing
+
+import (
+	"sync"
+	"time"
+
+	clockapi "k8s.io/utils/clock"
```
Comment (apelisse) [inline clock/testing/fake_clock.go]: Why not just clock?

File: clock/testing/fake_clock_test.go (hunk 1 above)
Comment (dashpole) [reply clock/testing/fake_clock_test.go]: yeah, we are just using golang's time library

**Comment by apelisse** [comment]
Thank you for taking care of that that quickly

File: clock/testing/fake_clock.go (hunk 2 above)
Comment (dashpole) [reply clock/testing/fake_clock.go]: changed

**Comment by thockin** [comment]
Thanks!

**Pull request merged**